File: Day_Three/Morning-Session/python/like_system_test.py
#Test

from unittest import TestCase

import like_system 

class TestPostLikes(TestCase):
    
    def test_that_empty_list_returns_no_one_likes(self):
        
        array = []
        actual = like_system.collect_names(array)
        expected = "no one likes this"
        self.assertEqual(expected, actual)


    def test_that_names_lis_is_one_returns_names(self):
        
        array = ["Praise"]
        actual = like_system.collect_names(array)
        expected = "Praise likes this"
        self.assertEqual(expected, actual)

    def test_that_names_list_is_two_returns_names(self):
        
        array = ["Praise","Eob"]
        actual = like_system.collect_names(array)
        expected = "Praise and Eob likes this"
        self.assertEqual(expected, actual)

    def test_that_names_list_is_three_returns_names_and_comment(self):
        
        array = ["Praise","Eob","biscuit"]
        actual = like_system.collect_names(array)
        expected = "Praise, Eob and biscuit likes this"
        self.assertEqual(expected, actual)

File: Day_Three/Morning-Session/java/LikeSystemTest.java
import org.junit.jupiter.api.Test;


import static org.junit.jupiter.api.Assertions.*;

public class LikeSystemTest{

    @Test
    public void testThatFuctionReturnsEmptyArrayIfThereWereNoLikes(){
        
        String[] array = {};
        String expected = "no one likes this";
        String actual = LikeSystem.collectName(array);
        assertEquals(actual, expected);
    }

     @Test
    public void testThatFuctionReturnsNameofLikers(){
        
        String[] array = {"Praise"};
        String expected = "Praise likes this";
        String actual = LikeSystem.collectName(array);
        assertEquals(actual, expected);
    }

     @Test
    public void testThatFuctionReturnsNameofLikersForTwoPeople(){
        
        String[] array = {"Praise","eob"};
        String expected = "Praise and eob likes this";
        String actual = LikeSystem.collectName(array);
        assertEquals(actual, expected);
    }

     @Test
    public void testThatFuctionReturnsNameofLikersOfThreePeople(){
        
        String[] array = {"Praise","eob","Lamine"};
        String expected = "Praise , eob and Lamine likes this";
        String actual = LikeSystem.collectName(array);
        assertEquals(actual, expected);
    }

     @Test
    public void testThatFuctionReturnsNameofLikersOfFourPeople(){
        
        String[] array = {"Praise","eob","Lamine","semicolon"};
        String expected = "Praise , eob and 2 others likes this";
        String actual = LikeSystem.collectName(array);
        assertEquals(actual, expected);
    }

    

}
